modules/Individual Needs/investigations_manage_edit.php

```diff
@@ -0,0 +1,327 @@
+                    	$column->addTextArea('reason')->setRows(5)->setClass('fullWidth')->required()->readonly(!$canEdit || $investigation['status'] != 'Referral');
+
+                    //Strategies Tried
+                    $row = $form->addRow();
+                    	$column = $row->addColumn();
+                    	$column->addLabel('strategiesTried', __('Strategies Tried'));
+                    	$column->addTextArea('strategiesTried')->setRows(5)->setClass('fullWidth')->readonly(!$canEdit || $investigation['status'] != 'Referral');
+
+                    //Parents Informed?
+                    $row = $form->addRow();
+                        $row->addLabel('parentsInformed', __('Parents Informed?'));
+                        $row->addYesNo('parentsInformed')->selected('N')->required()->readonly(!$canEdit || $investigation['status'] != 'Referral');
+
+                    $form->toggleVisibilityByClass('parentsInformed')->onSelect('parentsInformed')->when('Y');
+
+                    //Parent Response
+                    $row = $form->addRow()->addClass('parentsInformed');
+                    	$column = $row->addColumn();
+                    	$column->addLabel('parentsResponse', __('Parent Response'));
+                    	$column->addTextArea('parentsResponse')->setRows(5)->setClass('fullWidth')->readonly(!$canEdit || $investigation['status'] != 'Referral');
+
+                    //Form Tutor Resolution
+                    if ($investigation['status'] == 'Resolved' || ($investigation['status'] == 'Referral' && $isTutor)) {
+                        $form->addRow()->addHeading(__('Form Tutor Resolution'));
+                        if ($isTutor && $investigation['status'] == 'Referral') {
+                            $row = $form->addRow();
+                                $row->addLabel('resolvable', __('Resolvable?'))->description(__('Is form tutor able to resolve without further input? If no, further investigation will be launched.'));
+                                $row->addYesNo('resolvable')->required()->placeholder();
+
+                                $form->toggleVisibilityByClass('resolutionDetails')->onSelect('resolvable')->when('Y');
+                        }
+
+                        $form->toggleVisibilityByClass('invitationDetails')->onSelect('resolvable')->when('N');
+
+                        //Resolvable by tutor
+                        $row = $form->addRow()->addClass('resolutionDetails');
+                            $column = $row->addColumn();
+                            $column->addLabel('resolutionDetails', __('Resolution Details'));
+                            $column->addTextArea('resolutionDetails')->setRows(5)->setClass('fullWidth')->readonly(!$isTutor || $investigation['status'] != 'Referral');
+
+                        //Not resolvable by tutor
+                        try {
```

Comment:
Could these be moved into a Gateway? We're aiming to remove SQL from display and process scripts.

Reply:
Great suggestion: we've made the change and committed.